File: src/main/java/gargoyle/xenox/util/applet/JApplet.java
package gargoyle.xenox.util.applet;

import javax.swing.JPanel;
import java.net.URL;

public class JApplet extends JPanel implements Applet {
    private AppletStub stub;

    protected URL getCodeBase() {
        return stub.getCodeBase();
    }

    protected URL getDocumentBase() {
        return stub.getDocumentBase();
    }

    protected void setStub(AppletStub stub) {
        this.stub = stub;
    }
}

File: src/main/kotlin/gargoyle/xenox/util/applet/JApplet.kt
package gargoyle.xenox.util.applet

import java.net.URL
import javax.swing.JPanel

open class JApplet : JPanel(), Applet {
    private lateinit var stub: AppletStub
    protected val codeBase: URL?
        get() = stub.codeBase
    protected val documentBase: URL?
        get() = stub.documentBase

    fun setStub(stub: AppletStub) {
        this.stub = stub
    }
}
